src/com/oltpbenchmark/Phase.java

```diff
@@ -126,108 +91,124 @@ public boolean isThroughputRun() {
     }
 
     public void resetSerial() {
-        if (this.serial) {
-            this.nextSerials.clear();
-            this.nextSerials.addAll(this.startSerials);
-        }
+        this.nextSerial = 1;
```

Comment:
IIRC we restrict serial to only have 1 terminal anyway, where the old code was trying to let multiple terminals progress serially.